crates/rpc/rpc-eth-types/src/logs_utils.rs

```diff
@@ -129,9 +134,15 @@ pub fn append_matching_block_logs<P: BlockReader>(
                 all_logs.push(log);
             }
             log_index += 1;
+            // Check if we have reached the max logs per response and return the last block if so
+            if let Some(max_logs_per_response) = max_logs_per_response {
+                if all_logs.len() > max_logs_per_response {
+                    return Ok(Some(block_num_hash.number.saturating_sub(1)));
+                }
+            }
```

Comment:
I find this check a bit out of place here because we  now only need this to get the last block hash.

for the purpose of this check, we can move this to 

https://github.com/paradigmxyz/reth/pull/12790/files#diff-d45b799806905149d455157b0d93125d80d302c75be6c5c0f84882a45ec9ed6bR522 entirely because we can get the same info from the last log in all_logs then we don't need to apply this check per log

Reply:
Moving the check outside would mean we'd always process the entire block range, even when we know early on that the result will exceed the response size limit?

Reply:
good point
then lets move this to block scope